File: pascals-triangle-2.rb
# Solution to: https://leetcode.com/problems/pascals-triangle-ii/

def get_item(row_array, i)
  if i < 0 || i >= row_array.length
    return 0
  else
    row_array[i]
  end
end

def get_row(row_index)
  prev_row = [1]
  
  for i in 0..row_indexe
    puts 'I am i:'
    puts i
    new_row = []
    for j in 0..i
      new_num = get_item(prev_row, j-1) + get_item(prev_row, j)
      new_row[j] = new_num
    end 
    prev_row = new_row
  end
  prev_row
end

File: pascals-triangle-2.js
// Solution to: https://leetcode.com/problems/pascals-triangle-ii/

const getItem = function(rowArray, column) {
  if (column < 0 || column >= rowArray.length) {
    return 0;
  }
  return rowArray[column]
}

const getRow = function(rowIndex) {
    let prevRow = [1]
    for (let i = 0; i < rowIndex; ++i) {
        let newRow = new Array(i+1)
        for (let j = 0;j <= i + 1; ++j) {
            let newNum = getItem(prevRow, j-1) + getItem(prevRow, j)
            newRow[j] = newNum            
        }
        prevRow = newRow
    }
    return prevRow
};

